### DISCUSSION TITLE
error of Google Sign-in authentication on Flutter web

### DISCUSSION THREAD

**Discussion by Frank1016** [serverpod/serverpod#2576] [Q&A] [answered]
I followed along the Google authentication tutorial here: https://medium.com/serverpod/integrating-google-sign-in-with-serverpod-authentication-part-2-6fade3099baf

My google_client_secret.json file looks like this:
![image](https://github.com/user-attachments/assets/c9fa1f0a-da85-4282-83e1-08517b6e86e9)
and the SignInWithGoogleButton is like this:
![image](https://github.com/user-attachments/assets/57fd5ced-b051-48b0-854b-122029192955)

I didn't change anything regarding iOS or android part, as I was only testing it with Chrome.

After running the web app, I got the following error in terminal:
![image](https://github.com/user-attachments/assets/182d9f03-6086-499b-a5ff-fb34399e3153)

and the following is the error message in serverpod Insights:
![image](https://github.com/user-attachments/assets/36706679-58b8-4c12-9ad5-557092315bd2)

For serverClientId in SignInWithGoogleButton, I used the client_id value from google_client_secret.json file. I guess this may be wrong, but don't know where I can find this serverClientId value, as the tutorial is not clear enough for me.

Anyone can help to fix the issue? Thank you in advance!

**Comment by Isakdl** [comment]
Can you confirm that the google secret is located in `config/google_client_secret.json` in the server project?

The error message you have in insights says that the server is unable to load the secret, and without that file it cannot complete the authentication.

**Reply by Frank1016** [reply]
Yes, it's there:
![image](https://github.com/user-attachments/assets/b13d8aa6-c7f4-485a-ba84-48db674b22cb)

**Reply by SandPod** [reply, marked as answer]
Unfortunately our secret loader will fail if any of the required lines are missing.

Please ensure that all of these fields can be loaded from the file: https://github.com/serverpod/serverpod/blob/b595b1a89f8591fb3fc1d57fe59a1e938212be76/modules/serverpod_auth/serverpod_auth_server/lib/src/business/google_auth.dart#L75-L98


We have an issue here to improve the error messages received: https://github.com/serverpod/serverpod/issues/840

**Reply by Frank1016** [reply]
I didn't change anything, I just tried again today, and suddenly everything starts working, which is really weird!

**Reply by SandPod** [reply]
I ran through the google authentication setup today and noticed that the client secret needs to be downloaded after the redirect Uris are configured. Otherwise they won't be part of the client secret and the setup will fail. We will add better validation on this in the tracking issue.
